core/node/da_clients/src/eigen/client.rs

```diff
@@ -1,46 +1,49 @@
-use std::{str::FromStr, sync::Arc};
+use std::str::FromStr;
 
-use async_trait::async_trait;
-use secp256k1::SecretKey;
+use eigenda_client_rs::{client::GetBlobData, EigenClient};
 use subxt_signer::ExposeSecret;
 use zksync_config::{configs::da_client::eigen::EigenSecrets, EigenConfig};
 use zksync_da_client::{
     types::{DAError, DispatchResponse, InclusionData},
     DataAvailabilityClient,
 };
 
-use super::sdk::RawEigenClient;
 use crate::utils::to_retriable_da_error;
 
-#[async_trait]
-pub trait GetBlobData: Clone + std::fmt::Debug + Send + Sync {
-    async fn call(&self, input: &str) -> anyhow::Result<Option<Vec<u8>>>;
-}
-
-/// EigenClient is a client for the Eigen DA service.
+// We can't implement DataAvailabilityClient for an outside struct, so it is needed to defined this intermediate struct
 #[derive(Debug, Clone)]
-pub struct EigenClient<T: GetBlobData> {
-    pub(crate) client: Arc<RawEigenClient<T>>,
+pub struct EigenClientProxy<T: GetBlobData> {
+    client: EigenClient<T>,
```

Comment:
Maybe rename the struct `EigenClientProxy` to `EigenClient`, and redefine the import as `RawEigenClient`